daily_milestone.go

```diff
@@ -158,34 +175,47 @@ func getMilestones(baseURL string, token string, projectID string) ([]simpleMile
 func createMilestoneData(advance int, timeInterval string) []simpleMilestone {
 	today := time.Now().Local()
 	list := []simpleMilestone{}
-	for i := 0; i < advance; i++ {
-		if timeInterval == "daily" {
+
+	switch {
+	case timeInterval == "daily":
+		for i := 0; i < advance; i++ {
 			date := today.AddDate(0, 0, i).Format("2006-01-02")
 			milestone := simpleMilestone{}
 			milestone.Title = date
 			milestone.DueDate = date
 			list = append(list, milestone)
-		} else if timeInterval == "weekly" {
-			today = today.AddDate(0, 0, 7)
-			year, week := today.ISOWeek()
-			milestoneYear := strconv.Itoa(year)
-			milestoneWeek := strconv.Itoa(week)
+		}
+	case timeInterval == "weekly":
+		lastDay := lastDayWeek(today)
+		milestone := simpleMilestone{}
+		year, week := lastDay.ISOWeek()
+		milestone.Title = strconv.Itoa(year) + "-w" + strconv.Itoa(week)
+		milestone.DueDate = lastDay.Format("2006-01-02")
+		list = append(list, milestone)
```

Comment:
This line is the "offender" concerning my comment. It will result in the lastDay of the previous week to be added. And then within the for loop later it will add 7 days for the same week and be correct and add a milestone with the same title.

Reply:
Ok So I get slices out of bound error on same code and same number of milestones and I have no idea why. It runs sometimes and sometimes it doesn't.

Reply:
The issue was that not all old milestones were returned as pagination only returned a subset of the old milestones. I'll push some changes that get 100 milestones back for now. We should implement pagination later so.